### DISCUSSION TITLE
Solved: ImageMagick always adds green background

### DISCUSSION THREAD

**Discussion by GWGDevelop** [ImageMagick/ImageMagick#7786] [Help]
### ImageMagick version

7.0.10-35 Q16 x64

### Operating system, version and so on

Windows 10, 11, Server 2022 different versions

### Description

We're using ImageMagick on several computers to resize and convert differnt pictures to the same file format. It's exactly the same version of image magick on all computers.
On few computers this conversion adds som strange green background to the picture. For a while I accused my users of using a wrong conversion command.
But now I've opened the same image file with IMDisplay on two different computers. The original looks like this:
![image](https://github.com/user-attachments/assets/3b9c09bb-284b-44ff-8e86-ac0782c37fd2)

But on a few computer IMDisplay shows this:
![image](https://github.com/user-attachments/assets/53c039f4-f00a-4e8b-83f0-1e835ef47e53)

I have searched IM help and discussions but all I found was about how to add a background, not how NOT to have a background added automatically.
I'm sure, it must be some minor setting, but I can't seem to find it.

**Comment by snibgo** [comment]
It looks as if the tones and colours have been reversed. Please post, or link to, the input image.

**Reply by GWGDevelop** [reply]
Both images shown in my original post ARE the original input image. It is the same file - without conversions - opened with the IMDisplay.exe program. On one computer it shows correctly (brown with white background) on another computer it is displayed green with a green background.
If you like, you can download the file here: https://files.tailorsoft.de/raastof.jpg
But I'm sure, it's not the file, but it must be something on that one computer.
The operating System is Windows Server 2022 Datacenter 21H2, ImageMagick is 7.0.10-35 Q16 x64

**Comment by fmw42** [comment]
If the input and output is CMYK, that might be the source of the issue.  Some viewers do not view CMYK properly.  Post the input and identify the exact OS. Also provide the output from

`magick -version`

**Reply by GWGDevelop** [reply]
The viewer is, as I wrote, the IMDisplay.exe viewer that comes with ImageMagick. 
The output file looks exactly the same way as the UNCONVERTED preview of the original image in IMViewer.exe (except that it's smaller).

**Reply by fmw42** [reply]
What was your exact command line from input to output that you then viewed?

**Reply by fmw42** [reply]
The file is CMYK and has two clip paths.  Not sure what IM does with two? Looks like on one system the conversion or viewer failed in some way and is displaying the CMYK as sRGB.  There is a profile also and perhaps the profile is missing on those systems. That might cause the issue if IM does not have the CMYK profile to use or IMdisplay can't find it.  Check that   Coated FOGRA39 (ISO 12647-2:2004) is available on each system.

**Comment by snibgo** [comment]
In Windows 11, IMDisplay 1.0 with ImageMagick 7.1.1-20, the image displays with no obvious problem. I suspect that IMDisplay ignores the embedded CMYK profile, and uses a simple "`-colorspace sRGB`" for conversion.

The OP version of IM is older than mine. I think IMDisplay is no longer supplied.

**Reply by GWGDevelop** [reply]
I tried to install the newer IM version 7.1.1-20, and, as you already found out, it works and shows the picture correctly on every computer I tried.
Thanks for that!
... but... now I have the next problem: for automatic conversion I have a program that uses the ImageMagick OLE Control. This OLE Control is no longer part of the 7.1.1-20 Installation.
Do you have any Idea, how I can install the OLE Control for a newer IM version?

**Comment by snibgo** [comment]
About OLE control, see https://github.com/ImageMagick/ImageMagick/discussions/4913 .

**Reply by GWGDevelop** [reply]
Yes, thank you. I had just found those threads. Unfortunately the alternative would be, finding out, how to implement the .NET Library in Delphi, building a new wrapper for it and restructuring 5 Applications that implement the old OLE Wrapper. That's at least one complete week of reading, understanding, reprogramming.
My newest IM version Setup that contains the OLE Control is 7.0.10. 
I found out that 7.0.19 is the latest that brings the OLE  Control. 
Does anybody know, where to find those Setup EXE Files? I could only find the .tar.gz with the sources, and also finding out, what to install, so I can compile it myself would take me several days.

... maybe I can find a different way to get it to correctly read and convert correctly an all our computers. 
What is that "FOGRA39 (ISO 12647-2:2004)" - how can I install it seperately? Maybe that will solve my problems?

**Comment by GWGDevelop** [comment]
solved: minor version differences.
Now I've searched and tried a while with different setups I could find in different backups I made over the years.
It seems, I had installed version 7.0.10-61 on some computers, and 7.0.10-35 on the ones, where it went wrong. 
Now I found out, I can install the 7.0.10-35 and afterwards overwrite all files with the -65 version files and everything works like it should.
Sorry for the hassle and thanks for all your help!

**Comment by snibgo** [comment]
> What is that "FOGRA39 (ISO 12647-2:2004)" - how can I install it seperately? Maybe that will solve my problems?

Computer screens use red, green and blue colours. Often, the colours are defined by the sRGB standard, but some other standard may be used, such as DisplayP3 or AdobeRGB.

When an image has an embedded ICC profile (as yours does), the most accurate conversion to RGB colours is to use the appropriate profile. For example:

```
magick raastof.jpg -profile sRGB.icc x.jpg

magick raastof.jpg -profile P3D65.icc x.jpg

magick raastof.jpg -profile AdobeRGB1998.icc x.jpg
```
sRBG.icc is distributed with ImageMagick.
